wgpu/gui/qt.py

```diff
@@ -147,12 +216,19 @@ class QWgpuCanvas(WgpuCanvasBase, QtWidgets.QWidget):
     # detect this. See https://github.com/pygfx/wgpu-py/pull/68
 
     def __init__(self, *, size=None, title=None, max_fps=30, **kwargs):
+        # When using Qt, there needs to be an
+        # application before any widget is created
+        global app
+        if app is None:
+            app = QtWidgets.QApplication.instance() or QtWidgets.QApplication([])
```

Comment:
We could also create a function:
```py
def get_app():
   return QtWidgets.QApplication.instance() or QtWidgets.QApplication([])
```

And call that here and in `run()`. That way we don't need that global variable, and `run()` would also work if no widgets are created yet (e.g. in a use case that dynamically creates canvases).